numpyro/infer/inspect.py

```diff
@@ -359,21 +361,40 @@ def _resolve_plate_samples(plate_samples):
     }
 
     def get_log_probs(sample):
+        class substitute_deterministic(handlers.substitute):
+            def process_message(self, msg):
+                if msg["type"] == "deterministic":
+                    msg["args"] = (msg["value"],)
+                    msg["kwargs"] = {}
+                    msg["value"] = self.data.get(msg["name"])
+                    msg["fn"] = lambda x: x
```

Comment:
nit: this is required when self.data does not include the deterministic value (no substitute). You provided deterministic inputs in `samples` below so no need to have this logic. Keeping this as-is is fine btw.